### PR TITLE
Subscription: avoid the file reading when the memory is not enough for tsfile slicing & implement all-out effort global timeout control & support client resume from breakpoint for tsfile consumption

### PR DESCRIPTION
**Pull request by VGalaxies**
follow up #13954

### CODE REVIEW

File: iotdb-client/session/src/main/java/org/apache/iotdb/session/subscription/consumer/SubscriptionExecutorServiceManager.java [hunk 1]
```diff
@@ -285,10 +284,12 @@ int getAvailableCount() {
       if (!isShutdown()) {
         synchronized (this) {
           if (!isShutdown()) {
-            return Math.max(
-                ((ThreadPoolExecutor) this.executor).getPoolSize()
-                    - ((ThreadPoolExecutor) this.executor).getActiveCount(),
-                0);
```
Comment (VGalaxies) [inline iotdb-client/session/src/main/java/org/apache/iotdb/session/subscription/consumer/SubscriptionExecutorServiceManager.java:291]: alway zero before...

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/task/execution/SubscriptionSubtaskExecutor.java [hunk 2]
```diff
@@ -20,14 +20,64 @@
...
+  protected PipeSubtaskScheduler schedulerSupplier(final PipeSubtaskExecutor executor) {
+    return new SubscriptionSubtaskScheduler((SubscriptionSubtaskExecutor) executor);
+  }
+
+  public void executeReceiverSubtask(
+      final SubscriptionReceiverSubtask subtask, final long timeoutMs) throws Exception {
+    if (!super.hasAvailableThread()) {
+      subtask.call(); // non-strict timeout
+      return;
+    }
```
Comment (VGalaxies) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/task/execution/SubscriptionSubtaskExecutor.java:60]: alway this branch now

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/tsfile/PipeTsFileInsertionEvent.java [hunk 3]
```diff
@@ -496,30 +501,41 @@ public Iterable<TabletInsertionEvent> toTabletInsertionEvents() {
...
+            "Wait for resource enough for parsing {} for {} seconds.",
+            resource != null ? resource.getTsFilePath() : "tsfile",
+            waitTimeSeconds);
+      }
+
+      if (waitTimeSeconds * 1000 > timeoutMs) {
+        // should contain 'TimeoutException' in exception message
+        throw new InterruptedException(
+            String.format("TimeoutException: Waited %s seconds", waitTimeSeconds));
+      }
```
Comment (VGalaxies) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/tsfile/PipeTsFileInsertionEvent.java:538]: 10s print once
finally print

Comment (VGalaxies) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/tsfile/PipeTsFileInsertionEvent.java:538]: 10s print once
finally print

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**